diesel/src/upsert/on_conflict_extension.rs

```diff
@@ -95,16 +97,34 @@ where
     /// ### Specifying a column as the target
     ///
     /// ```rust
-    /// # include!("on_conflict_docs_setup.rs");
+    /// # include!("../doctest_setup.rs");
     /// #
-    /// # #[cfg(any(feature = "sqlite", feature = "postgres"))]
+    /// # table! {
+    /// #     users {
+    /// #         id -> Integer,
+    /// #         name -> VarChar,
+    /// #         hair_color -> VarChar,
+    /// #     }
+    /// # }
+    /// #
+    /// # #[derive(Clone, Copy, Insertable)]
+    /// # #[diesel(table_name = users)]
+    /// # struct User<'a> {
+    /// #     id: i32,
+    /// #     name: &'a str,
+    /// # }
+    /// #
+    /// # #[cfg(any(feature = "sqlite", feature = "postgres", feature = "mysql"))]
     /// # fn main() {
     /// #     use self::users::dsl::*;
+    /// use diesel::upsert::*;
+    ///
     /// #     let conn = &mut establish_connection();
-    /// #     #[cfg(feature = "postgres")]
-    /// #     diesel::sql_query("TRUNCATE TABLE users").execute(conn).unwrap();
-    /// #     #[cfg(feature = "sqlite")]
-    /// #     diesel::sql_query("DELETE FROM users").execute(conn).unwrap();
+    /// #     diesel::sql_query("DROP TABLE users").execute(conn).unwrap();
+    /// #[cfg(any(feature = "sqlite", feature = "postgres"))]
+    /// #     diesel::sql_query("CREATE TABLE users (id SERIAL PRIMARY KEY, name TEXT)").execute(conn).unwrap();
+    /// #[cfg(feature = "mysql")]
+    /// #     diesel::sql_query("CREATE TABLE users (id SERIAL PRIMARY KEY, name VARCHAR(255))").execute(conn).unwrap();
```

Comment:
Note that this (and the custom doctest setup above) is needed because MySQL doesn't allow unique indexes on `TEXT` columns -- ref [here](https://stackoverflow.com/questions/1827063/mysql-error-key-specification-without-a-key-length) on stackoverflow

Reply:
I think it might be fine to just use `VARCHAR` for the other backends as well.